boto3/s3/transfer.py

```diff
@@ -122,499 +122,71 @@ def __call__(self, bytes_amount):
-            during read() calls.
-
-        :rtype: ``ReadFileChunk``
-        :return: A new instance of ``ReadFileChunk``
-
-        """
-        f = open(filename, 'rb')
-        file_size = os.fstat(f.fileno()).st_size
-        return cls(f, start_byte, chunk_size, file_size, callback,
-                   enable_callback)
-
-    def _calculate_file_size(self, fileobj, requested_size, start_byte,
-                             actual_file_size):
-        max_chunk_size = actual_file_size - start_byte
-        return min(max_chunk_size, requested_size)
-
-    def read(self, amount=None):
-        if amount is None:
-            amount_to_read = self._size - self._amount_read
-        else:
-            amount_to_read = min(self._size - self._amount_read, amount)
-        data = self._fileobj.read(amount_to_read)
-        self._amount_read += len(data)
-        if self._callback is not None and self._callback_enabled:
-            self._callback(len(data))
-        return data
-
-    def enable_callback(self):
-        self._callback_enabled = True
-
-    def disable_callback(self):
-        self._callback_enabled = False
-
-    def seek(self, where):
-        self._fileobj.seek(self._start_byte + where)
-        if self._callback is not None and self._callback_enabled:
-            # To also rewind the callback() for an accurate progress report
-            self._callback(where - self._amount_read)
-        self._amount_read = where
-
-    def close(self):
-        self._fileobj.close()
-
-    def tell(self):
-        return self._amount_read
-
-    def __len__(self):
-        # __len__ is defined because requests will try to determine the length
-        # of the stream to set a content length.  In the normal case
-        # of the file it will just stat the file, but we need to change that
-        # behavior.  By providing a __len__, requests will use that instead
-        # of stat'ing the file.
-        return self._size
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, *args, **kwargs):
-        self.close()
-
-    def __iter__(self):
-        # This is a workaround for http://bugs.python.org/issue17575
-        # Basically httplib will try to iterate over the contents, even
-        # if its a file like object.  This wasn't noticed because we've
-        # already exhausted the stream so iterating over the file immediately
-        # stops, which is what we're simulating here.
-        return iter([])
-
-
-class StreamReaderProgress(object):
-    """Wrapper for a read only stream that adds progress callbacks."""
-    def __init__(self, stream, callback=None):
-        self._stream = stream
-        self._callback = callback
 
-    def read(self, *args, **kwargs):
-        value = self._stream.read(*args, **kwargs)
-        if self._callback is not None:
-            self._callback(len(value))
-        return value
+class TransferConfig(S3TransferConfig):
```

Comment:
I think long term (i.e. once s3transfer hits 1.0) they will not use ``S3Transfer`` or ``TransferConfig`` at all. I would prefer to slap a big deprecated flag on the entire module and point them to using ``s3transfer`` directly. But for now since this is not the case, I would prefer that we cover up ``S3TransferConfig`` by having them go through ``TransferConfig`` to allow us to make changes if needed to the ``s3transfer.manager.TransferConfig`` interface. Does that make sense?